stream2hop/tns2hop.py

```diff
@@ -0,0 +1,254 @@
+import argparse
+from . import Utils as ut
+from twisted.internet import task, reactor
+from datetime import datetime, timedelta
+import requests
+import json
+import wget
+import csv
+import os
+
+
+def _add_parser_args(parser):
+    """
+        adding parser arguments
+    """
+    ut.add_common_arguments(parser)
+    parser.add_argument("--api_key", help="TNS Bot's api key")
+
+
+def search(url, json_list, api_key):
+    """
+      Search for objects in TNS
+
+      Args:
+        url: TNS url
+        json_list: parameters in the search query
+        api_key: TNS bot's api key
+
+      Return:
+        search result
+    """
+    try:
+        # url for search obj
+        search_url = url + "/search"
```

Comment:
I would use `urllib.parse.urljoin` everywhere you're combining URLs like these to ensure they always resolve to valid URLs: https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urljoin

Reply:
I will apply this change to all URLs concatenations